pkg/virt-launcher/virtwrap/converter/converter_test.go

```diff
@@ -3430,3 +3437,19 @@ func False() *bool {
 	b := false
 	return &b
 }
+
+// As the arch specific default disk is set in the mutating webhook, so in some tests,
+// it needs to run the mutate function before verifying converter
+func vmiArchMutate(arch string, vmi *v1.VirtualMachineInstance, c *ConverterContext) {
+	if arch == "arm64" {
+		webhooks.SetVirtualMachineInstanceArm64Defaults(vmi)
+		// EFI related setting is done in the mutator for Arm64, add EFIConfiguration
+		// in the ConverterContext
+		c.EFIConfiguration = &EFIConfiguration{
```

Comment:
It is not entirely clera to me why we have to set this on the converter for arm64 but not for amd64.

Reply:
In `SetVirtualMachineInstanceArm64Defaults`, we do `setDefaultBootloader` which initialize the Bootloader. This leads to, in the function `Convert_v1_VirtualMachineInstance_To_api_Domain`, it need a value form `&c.EFIConfiguration.SecureLoader` to detect if secureboot is applied.
https://github.com/kubevirt/kubevirt/blob/6fab41846968b8eddc7f5126c47f1c211e59a98f/pkg/virt-launcher/virtwrap/converter/converter.go#L1230-L1236
Without setting it, on arm64, it would get an error ` invalid memory address or nil pointer dereference`.

However, on amd64, we do not initialize the Bootloader in `SetVirtualMachineInstanceAmd64Defaults`.

Reply:
ack.